AVOutput/AVOutputTV.cpp

```diff
@@ -0,0 +1,5263 @@
+* limitations under the License.
+*/
+
+#include <string>
+#include "AVOutputTV.h"
+#include "UtilsIarm.h"
+#include "rfcapi.h"
+
+#define BUFFER_SIZE     (128)
+
+#define registerMethod(...) for (uint8_t i = 1; GetHandler(i); i++) GetHandler(i)->Register<JsonObject, JsonObject>(__VA_ARGS__)
+
+static bool filmMakerMode= false;
+static std::map<std::string, int> supportedSourcemap;
+static std::map<std::string, int> supportedPictureModemap;
+static std::map<std::string, int> supportedFormatmap;
+static bool m_isDalsEnabled = false;
+
+namespace WPEFramework {
+namespace Plugin {
+
+    AVOutputTV* AVOutputTV::instance = nullptr;
+
+    static void tvVideoFormatChangeHandler(tvVideoFormatType_t format, void *userData)
+    {
+        LOGINFO("tvVideoFormatChangeHandler format:%d \n",format);
+        AVOutputTV *obj = (AVOutputTV *)userData;
+        if(obj)obj->NotifyVideoFormatChange(format);
+    }
+
+    static void tvFilmMakerModeChangeHandler(tvContentType_t mode, void *userData)
+    {
+        LOGINFO("tvFilmMakerModeChangeHandler content:%d \n",mode);
+        AVOutputTV *obj = (AVOutputTV *)userData;
+        if(obj)obj->NotifyFilmMakerModeChange(mode);
+    }
+
+    static void tvVideoResolutionChangeHandler(tvResolutionParam_t resolution, void *userData)
+    {
+        LOGINFO("tvVideoResolutionChangeHandler resolution:%d\n",resolution.resolutionValue);
+        AVOutputTV *obj = (AVOutputTV *)userData;
+        if(obj)obj->NotifyVideoResolutionChange(resolution);
+    }
+
+    static void tvVideoFrameRateChangeHandler(tvVideoFrameRate_t frameRate, void *userData)
+    {
+        LOGINFO("tvVideoFrameRateChangeHandler format:%d \n",frameRate);
+        AVOutputTV *obj = (AVOutputTV *)userData;
+        if(obj)obj->NotifyVideoFrameRateChange(frameRate);
+    }
+
+    static bool getVideoContentTypeToString(tvContentType_t content)
+    {
+        bool fmmMode = false;
+        switch(content) 
+	{
+            case tvContentType_FMM:
+                LOGINFO("Content Type: FMM\n");
+                fmmMode = true;
+                break;
+            default:
+                LOGINFO("Content Type: NONE\n");
+                fmmMode = false;
+                break;
+        } 
+        return fmmMode;
+    }
+
+    static const char *getVideoFormatTypeToString(tvVideoFormatType_t format)
+    {
+        const char *strValue = "NONE";
+        switch(format) 
+	{
+            case VIDEO_FORMAT_SDR:
+                LOGINFO("Video Format: SDR\n");
+                strValue = "SDR";
+                break;
+            case VIDEO_FORMAT_HDR10:
+                LOGINFO("Video Format: HDR10\n");
+                strValue = "HDR10";
+                break;
+            case VIDEO_FORMAT_HDR10PLUS:
+                LOGINFO("Video Format: HDR10PLUS\n");
+                strValue = "HDR10PLUS";
+                break;
+            case VIDEO_FORMAT_HLG:
+                LOGINFO("Video Format: HLG\n");
```

Comment:
instrad of repeating in every case LOGINFO("Video Format: :%s \n",strValue ); at end

Reply:
done